ContainerLogsV2 and Basic Logs support

samaea opened the pull request:
## PR Summary

Addresses issue https://github.com/Azure/AKS-Construction/issues/437. 

In addition, added:-

- updateDependencyFn() function (under addonsTabs) that assists with enforcing dependency between two features (e.g. In order to activate Basic Logs on Container Logs V2, you also need to enable the Container Logs V2 schema. It might be worth considering to make this a global function instead (i.e same place as where updateFn() is defined).
- Under deployTabs, preview_post_params array has been added to support flagging preview features passed as params to the postdeploy script. At present, we are only capturing preview params under ARM/Bicep.

![image](https://user-images.githubusercontent.com/32162601/211028976-9a95e63f-a31d-424d-be10-033f1be3f169.png)

![image](https://user-images.githubusercontent.com/32162601/211030504-2c7abaf1-4e0b-41d1-aa88-99f35276db16.png)

## PR Checklist

- [X] PR has a meaningful title
- [X] Summarized changes
- [X] This PR is ready to merge and is not **Work in Progress**
- [X] Link to a filed issue
- [X] Screenshot of UI changes (if PR includes UI changes)

Gordonby commented on bicep/main.bicep:
```diff
@@ -1514,6 +1517,17 @@ resource aks_law 'Microsoft.OperationalInsights/workspaces@2022-10-01' = if (cre
+  properties: {
+    plan: 'Basic'
+  }
+  dependsOn: [
```
Do we need the log analytics table creation to wait for the AKS cluster?
I'd hope it's not connected, and we can drop the dependsOn.

Gordonby commented on postdeploy/scripts/certmanager-install.sh:
```diff
@@ -1,11 +0,0 @@
-#!/bin/bash
```
why has this file been removed?

samaea replied:
Great catch! Readded certmanager-install.sh as per offline discussion.

samaea commented:
👋 @khowling, incorporated feedback from our PR discussion. Also added another function in addons.js to ensure any features dependent on the ContainerLogsV2 schema are disabled as well (at present it only accommodates for the Basic Logs feature). This avoids the scenario where the user selects on Basic Logs (which automatically selects the ContainerLogsV2 feature) and then decides to disable the ContainerLogsV2 feature (i.e safety guard in both directions now).

khowling commented:
hey @samaea , looking good!  I just tested the UI, and one small thing,  in Monitoring, If I select 'Azure Monitor for Containers', then select the V2 Logs, the parameters appear on the deployment command as expected. But it I then select 'None' for Monitoring, the V2 log parameters still appear.  To fix this, can you add an additional check in the deployment logic, take a look at this as an example 
![image](https://user-images.githubusercontent.com/1034202/212676684-7f37abf9-bc0a-4688-bc02-b7cfa9bd5a74.png)

samaea commented:
> hey @samaea , looking good! I just tested the UI, and one small thing, in Monitoring, If I select 'Azure Monitor for Containers', then select the V2 Logs, the parameters appear on the deployment command as expected. But it I then select 'None' for Monitoring, the V2 log parameters still appear. To fix this, can you add an additional check in the deployment logic, take a look at this as an example ![image](https://user-images.githubusercontent.com/1034202/212676684-7f37abf9-bc0a-4688-bc02-b7cfa9bd5a74.png)

Hey @khowling - great catch! All done. :)
